src/components/SearchReps.vue

```diff
@@ -188,17 +175,43 @@ export default {
             this.step2 = step2
         },
         CheckInputContent: function () {
-            if (this.postalCode !== '') {
+            if (this.searchText !== '') {
                 this.hasContent = true
             } else {
                 this.hasContent = false
             }
         },
         async CreateRepList() {
             try {
-                const res = await axios.get(
-                    '/api/representatives/' + this.postalCode
+
+                // check postal code is valid with regex
+                let res = ''
+                let isPostalCodeValid =  /(^\d{5}$)|(^\d{5}-\d{4}$)/.test(this.searchText);
+                let streetAddressValid = /^[a-zA-Z0-9\s,'-]*$/.test(this.searchText);
```

Comment:
I know you mentioned wanting to improve this, this site allows you to search for common regex patterns: [link](https://regexlib.com/Search.aspx?k=us+street+address&c=-1&m=-1&ps=20)

I linked a search for US street address, but I also found results for postal code.  I wouldn't worry about getting something that matches everything exhaustively, but rather something that works "well enough" - the cicero API just won't return results if it doesn't find any for the given address, so we can put a note on the page for the user to check the address they entered and look for typos or something similar

Reply:
Thanks!